packaging/ubuntu-14.04/tests/integrationtests

```diff
@@ -23,6 +23,14 @@ sed -i 's/\(PermitRootLogin\|PasswordAuthentication\)\>.*/\1 yes/' /etc/ssh/sshd
 
 service ssh restart
 
+# Map snapd deb package pockets to core snap channels. This is intended to cope
+# with the autopkgtest execution when testing packages from the different pockets
+if [ -n "$(apt list snapd | grep \\-proposed)" ]; then
```

Comment:
two very minor nits: one, I'd rather have this snippet in tests/lib somewhere than have it duplicated. Two, you can do
```sh
if apt -qq list snapd | grep -q -- -proposed; then ...
```
which uses the exit status of the `grep` to branch instead of matching the string, which seems nicer to me (the extra `-q`s are just for cleanliness).

*or*

you could
```sh
case "$(apt -qq list snapd)" in
    *proposed*)
        export SPREAD_CORE_CHANNEL=candidate
        ;;
    *updates*)
        export SPREAD_CORE_CHANNEL=stable
esac
```
which is back to doing string parsing, but avoids the double call to apt which would result in it being significantly faster.

Question: what happens if apt's output matches neither of these things?